lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -499,3 +514,22 @@ instance Arbitrary WalletName where
 instance Arbitrary BoundType where
     shrink = genericShrink
     arbitrary = genericArbitrary
+
+newtype NonZero a = NonZero a
+
+instance (Arbitrary a, Num a, Eq a) => Arbitrary (NonZero a) where
+    arbitrary = NonZero . restrict <$> arbitrary
+      where
+        restrict x = if x == 0 then 1 else x
+    shrink (NonZero x) =
+        map NonZero
+        . suggestOne
+        . filter (/= 0)
+        $ shrink x
+      where
+        suggestOne shrinks = if shrinks == [] then [1] else []
```

Comment:
Ever heard of QuickCheck's modifiers :grimacing: ?

-> https://hackage.haskell.org/package/QuickCheck-2.12/docs/Test-QuickCheck.html#t:NonZero